OMCompiler/SimulationRuntime/c/simulation/solver/events.c

```diff
@@ -288,44 +284,50 @@ void handleEvents(DATA* data, threadData_t *threadData, LIST* eventLst, double *
  *  \param [ref] [data]
  *  \param [ref] [threadData]
  *  \param [ref] [eventList]
- *  \return: first event of interval [oldTime, timeValue]
- *
- *  This function perform a root finding for interval = [oldTime, timeValue]
+ *  \param [in]  [time_left]
+ *  \param [in]  [values_left]
+ *  \param [in]  [time_right]
+ *  \param [in]  [values_right]
+ *  \return: first event of interval [time_left, time_right]
  */
-double findRoot(DATA* data, threadData_t *threadData, LIST *eventList)
+double findRoot(DATA* data, threadData_t* threadData, LIST* eventList, double time_left, double* values_left, double time_right, double* values_right)
 {
   TRACE_PUSH
 
-  double eventTime;
-  long event_id;
   LIST_NODE* it;
   fortran_integer i=0;
-  static LIST *tmpEventList = NULL;
+  static LIST tmpEventList = (LIST){NULL, NULL, sizeof(long), 0};
```

Comment:
Whoops, I didn't think of that. I thought `long` depends on the architecture but would still be known at compile time and that would be enough... I don't really understand why `sizeof(long)` is non static, but I'll accept that for now :sweat_smile: 

I only made it static so that we don't need to `malloc` and `free` at each call. I guess it would still be better for memory management if I just remove the `static`, but I'm open for suggestions!

Reply:
Actually, I didn't make it static, it was static already.

Reply:
>  I don't really understand why sizeof(long) is non static, but I'll accept that for now 

I am also in the 'I will accept it for now' category with you. I guess it has some technical reason.  

> I only made it static so that we don't need to malloc and free at each call. I guess it would still be better for memory management if I just remove the static, but I'm open for suggestions!

I thought something like that too. If I understood you correctly, you do not need to make it static to avoid malloc. I think the fact that the word static has different meanings in different contexts added to the confusion. After your changes, it is still indeed a "static array" but it is not a static function variable 🤷‍♂️.

Reply:
I think maybe it is a good idea to revise the other static variables in the function. If they are not actually intended to be "static" (think global) then they might be hiding some bugs in your code since their `initialization` is only done once. 

For example. if I am reading it correctly, on the first ever call to the `findRoot` fuction `states_left` is initialized to NULL. However, in every subsequent call the initialization is skipped and the variable will have whatever value it had when the last call to the function finished. 

You can pout a print statement after the initialization and see what get's printed in repeated calls.

Reply:
> I think maybe it is a good idea to revise the other static variables in the function. If they are not actually intended to be "static" (think global) then they might be hiding some bugs in your code since their `initialization` is only done once.
> 
> For example. if I am reading it correctly, on the first ever call to the `findRoot()` fuction `states_left` is initialized to NULL. However, in every subsequent call the initialization is skipped and the variable will have whatever value it had when the last call to the function finished.
> 
> You can put a print statement after the initialization and see what gets printed in repeated calls.

That part is on purpose actually. I want to keep the memory block for later calls since it's always the same size. The arrays are overwritten by input data with `memcpy` anyways so there should be no danger. The only downside is that the two arrays are never freed. A better approach would probably be to put these arrays into the `DATA` object somewhere, but I did not find a good spot quickly. :sweat_smile: